src/main/java/com/rabbitmq/http/client/Client.java

```diff
@@ -614,7 +665,11 @@ private String encodePathSegment(final String pathSegment) {
     return xs;
   }
 
-  private HttpComponentsClientHttpRequestFactory getRequestFactory(final URL url, final String username, final String password, final SSLConnectionSocketFactory sslConnectionSocketFactory, final SSLContext sslContext) throws MalformedURLException {
+  private HttpComponentsClientHttpRequestFactory getRequestFactory(final URL url,
+                                                                   final String username, final String password,
+                                                                   final SSLConnectionSocketFactory sslConnectionSocketFactory,
+                                                                   final SSLContext sslContext,
+                                                                   final HttpClientBuilderConfigurator configurator) throws MalformedURLException {
```

Comment:
IMO all this new complexity still convinces me that requesting just an `HttpClient` would be enough to cover all the users requirements.
Our mandatory header `HttpHeaders.CONTENT_TYPE, "application/json"` we would populate via `HttpEntity(T body, MultiValueMap<String, String> headers)`...